mlrun/db/httpdb.py

```diff
@@ -721,54 +722,100 @@ def del_runs(self, name=None, project=None, labels=None, state=None, days_ago=0)
         error = "del runs"
         self.api_call("DELETE", "runs", error, params=params)
 
-    def store_artifact(self, key, artifact, uid, iter=None, tag=None, project=""):
+    def store_artifact(
+        self,
+        key,
+        artifact,
+        # TODO: deprecated, remove in 1.7.0
+        uid=None,
+        iter=None,
+        tag=None,
+        project="",
+        tree=None,
+    ):
         """Store an artifact in the DB.
 
         :param key: Identifying key of the artifact.
         :param artifact: The actual artifact to store.
-        :param uid: A unique ID for this specific version of the artifact.
+        :param uid: A unique ID for this specific version of the artifact
+                    (deprecated, artifact uid is generated in the backend use `tree` instead)
         :param iter: The task iteration which generated this artifact. If ``iter`` is not ``None`` the iteration will
             be added to the key provided to generate a unique key for the artifact of the specific iteration.
         :param tag: Tag of the artifact.
         :param project: Project that the artifact belongs to.
+        :param tree: The tree which generated this artifact.
         """
+        if uid:
+            warnings.warn(
+                "'uid' is deprecated." "This will be removed in 1.7.0",
+                # TODO: Remove this in 1.7.0
+                FutureWarning,
+            )
```

Comment:
Add: `tree = tree or uid`
And add a comment that this is only because `uid` was used in the past as a terrible name for `tree` (which is not great itself). As discussed, we need to add a new type of `uid` that will indicate the actual uid of the resource. Maybe call it `object_uid`?